sdk/examples/xo_java/XoHandler.java

```diff
@@ -0,0 +1,445 @@
+/* Copyright 2017 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+------------------------------------------------------------------------------*/
+
+package sawtooth.examples.xo;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import org.apache.commons.lang3.StringUtils;
+
+import sawtooth.sdk.processor.State;
+import sawtooth.sdk.processor.TransactionHandler;
+import sawtooth.sdk.processor.Utils;
+import sawtooth.sdk.processor.exceptions.InternalError;
+import sawtooth.sdk.processor.exceptions.InvalidTransactionException;
+import sawtooth.sdk.protobuf.TpProcessRequest;
+import sawtooth.sdk.protobuf.TransactionHeader;
+
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.Type;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+public class XoHandler implements TransactionHandler {
+
+  private final Logger logger = Logger.getLogger(XoHandler.class.getName());
+  private String xoNameSpace;
+
+  /**
+   * constructor.
+   */
+  public XoHandler() {
+    try {
+      this.xoNameSpace = Utils.hash512(
+        this.transactionFamilyName().getBytes("UTF-8")).substring(0, 6);
+    } catch (UnsupportedEncodingException usee) {
+      usee.printStackTrace();
+      this.xoNameSpace = "";
+    }
+  }
+
+  @Override
+  public String transactionFamilyName() {
+    return "xo";
+  }
+
+  @Override
+  public String getEncoding() {
+    return "csv-utf8";
+  }
+
+  @Override
+  public String getVersion() {
+    return "1.0";
+  }
+
+  @Override
+  public Collection<String> getNameSpaces() {
+    ArrayList<String> namespaces = new ArrayList<>();
+    namespaces.add(this.xoNameSpace);
+    return namespaces;
+  }
+
+  @Override
+  public void apply(TpProcessRequest transactionRequest, State stateStore)
+      throws InvalidTransactionException, InternalError {
+    String[] transactionData = getUnpackedTransaction(transactionRequest);
```

Comment:
Instead of this being an area, this should probably just be a package-private class in this file, like so:

```
class TransactionData {
    final String action;
    final String gameName;
    final String space;

    TransactionData(String action, String gameName, String space) {
        this.action = action;
        this.gameName = gameName;
        this.space = space;
    }
}
```

Those fields can just be read directly, no need for getters, as their finals and private.